### PR TITLE
Minimap: show player markers

### PR DESCRIPTION
**Pull request by RealBadAngel**
screenshot_20160218_041932

### CODE REVIEW

**Comment by MoNTE48** [comment]
:+1:

**Comment by C1ffisme** [comment]
:+1: 

As long as servers can turn it off if needed.

**Comment by RealBadAngel** [comment]
@C1ffisme serveers already can disable minimap

**Comment by C1ffisme** [comment]
Okay. :+1:

**Comment by Fixer-007** [comment]
Squared minimap shows markers wrongly.

**Comment by RealBadAngel** [comment]
@Fixer-007 fixed

**Comment by Ferk** [comment]
Nice! ...even nicer if it's done in a generic way so in the future there can be an API for mods to add custom markers with custom textures.

File: src/minimap.cpp [hunk 1]
```diff
@@ -527,6 +534,70 @@ void Mapper::drawMinimap()
 	driver->setTransform(video::ETS_VIEW, oldViewMat);
 	driver->setTransform(video::ETS_PROJECTION, oldProjMat);
 	driver->setViewPort(oldViewPort);
+
+	// Draw player markers
+	v2s32 s_pos = v2s32(screensize.X - size - 10, 10);
```
Comment (nerzhul) [inline src/minimap.cpp]: no assignment, use constructor !

File: src/minimap.cpp [hunk 2]
```diff
@@ -527,6 +534,70 @@ void Mapper::drawMinimap()
...
+	core::dimension2di imgsize(data->object_marker_red->getOriginalSize());
+	core::rect<s32> img_rect = core::rect<s32>(0, 0, imgsize.Width, imgsize.Height);
```
Comment (nerzhul) [inline src/minimap.cpp]: same here

**Comment by nerzhul** [comment]
i think markers should be disabled without disable the whole minimap. It could be useful on servers to have a minimap without players, for pvp for example, without hardcore rules.

**Comment by paramat** [comment]
The marker textures are well made but i think a little excessive at 31x31 pixels, considering how small they appear on-screen, how about 15x15:
object_marker_red_15

**Comment by RealBadAngel** [comment]
@paramat, they do scale depending on screen resolution. ive made them a bit larger so you wont get pixelated marker with greater resolutions, they do match 512x512 original minimap size

**Comment by rubenwardy** [comment]
> @C1ffisme servers already can disable minimap

> i think markers should be disabled without disable the whole minimap. It could be useful on servers to have a minimap without players, for pvp for example, without hardcore **rules.**

I'd like this too.

**Comment by RealBadAngel** [comment]
About disabling anything by server, imho this cannot be part of this feature PR, see:
https://github.com/minetest/minetest/commit/be9024a397506dff1151b16e5f7a91e393b00d7e

**Comment by Fixer-007** [comment]
My issues are fixed. :+1:

**Comment by est31** [comment]
> i think markers should be disabled without disable the whole minimap. It could be useful on servers to have a minimap without players, for pvp for example, without hardcore rules.

@nerzhul  I am preparing a commit that makes minimap completely configurable from server side.

File: src/minimap.cpp [hunk 3]
```diff
@@ -527,6 +534,70 @@ void Mapper::drawMinimap()
...
+			s_pos.X + posf.X - marker_size2,
+			s_pos.Y + posf.Y - marker_size2,
+			s_pos.X + posf.X + marker_size2,
+			s_pos.Y + posf.Y + marker_size2);
+		driver->draw2DImage(data->object_marker_red, dest_rect,
+			img_rect, &dest_rect, &c[0], true);
+	}
+}
+
+void Mapper::updateActiveMarkers (void)
```
Comment (est31) [inline src/minimap.cpp]: no (void) required

Comment (est31) [inline src/minimap.cpp]: better do () only

File: src/minimap.cpp [hunk 4]
```diff
@@ -250,6 +251,8 @@ Mapper::Mapper(IrrlichtDevice *device, Client *client)
 
 	// Create player marker texture
 	data->player_marker = m_tsrc->getTexture("player_marker.png");
+	// Create object marker textures
```
Comment (est31) [inline src/minimap.cpp]: singular

File: src/minimap.cpp [hunk 5]
```diff
@@ -274,6 +277,8 @@ Mapper::~Mapper()
 	driver->removeTexture(data->heightmap_texture);
 	driver->removeTexture(data->minimap_overlay_round);
 	driver->removeTexture(data->minimap_overlay_square);
+	driver->removeTexture(data->object_marker_red);
+	driver->removeTexture(data->object_marker_green);
```
Comment (est31) [inline src/minimap.cpp]: You forgot this I tihnk

File: src/camera.h [hunk 6]
```diff
@@ -172,6 +172,9 @@ class Camera
 
 	void removeNametag(Nametag *nametag);
 
+	std::list<Nametag *> *getNametags()
+	{ return &m_nametags; }
```
Comment (est31) [inline src/camera.h]: trailing whitespace

**Comment by est31** [comment]
Merged: 5f4ed94d90668af58a3e677e7401f0028871acce

**Pull request closed without merging**